mne/time_frequency/psd.py

```diff
@@ -112,12 +111,209 @@ def _check_nfft(n, n_fft, n_overlap):
+        if proj:
+            inst = inst.apply_proj()
+
+        sfreq = inst.info['sfreq']
+        if isinstance(inst, _BaseRaw):
+            start, stop = inst.time_as_index([tmin, tmax])
+            data, times = inst[picks, start:(stop + 1)]
+        elif isinstance(inst, _BaseEpochs):
+            data = inst.get_data()[:, picks][..., time_mask]
+
+    elif isinstance(inst, np.ndarray):
+        if not isinstance(sfreq, (float, int)):
+            raise ValueError('If a numpy array is given, must provide sfreq')
+        data = inst.copy()
+        if data.ndim == 1:
+            data = data[np.newaxis, :]
+    else:
+        raise ValueError('epochs must be an instance of Epochs, Raw, '
+                         'or ndarray. Got type {0}'.format(type(inst)))
+    return data, sfreq
+
+
+@verbose
+def psd_welch(inst, fmin=0, fmax=np.inf, tmin=None, tmax=None, n_fft=256,
+              n_overlap=0, picks=None, proj=False, sfreq=None, n_jobs=1,
+              verbose=None):
+    """Compute the PSD using Welch's method.
+
+    Parameters
+    ----------
+    inst : instance of Epochs or Raw |
+           array of shape ([n_epochs], n_channels, n_times)
+        The data. If not an instance of Epochs or Raw,
+        sfreq must also be supplied.
+    fmin : float
+        Min frequency of interest
+    fmax : float
+        Max frequency of interest
+    tmin : float | None
+        Min time of interest
+    tmax : float | None
+        Max time of interest
+    n_fft : int
+        The length of the tapers ie. the windows. The smaller
+        it is the smoother are the PSDs. The default value is 256.
+        If ``n_fft > len(epochs.times)``, it will be adjusted down to
+        ``len(epochs.times)``.
+    n_overlap : int
+        The number of points of overlap between blocks. Will be adjusted
+        to be <= n_fft.
+        bandwidth : float
+        The bandwidth of the multi taper windowing function in Hz.
+    picks : array-like of int | None
+        The selection of channels to include in the computation.
+        If None, take all channels.
+    proj : bool
+        Apply SSP projection vectors. If inst is ndarray this is not used.
+    sfreq : float
+        The sampling frequency of the data. Required if inst is an array,
+        otherwise this is not used and sfreq is pulled from inst.
+    n_jobs : int
+        Number of CPUs to use in the computation.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    psds : ndarray, shape ([n_epochs], n_channels, n_freqs)
+        The power spectral densities. If Raw is provided,
+        then psds will be 2-D.
+    freqs : ndarray (n_freqs)
+        The frequencies.
+    """
+    from scipy.signal import welch
+
+    # Prep data
+    data, sfreq = _check_psd_data(inst, tmin, tmax, picks, proj, sfreq)
+    dshape = data.shape[:-1]
+    n_times = data.shape[-1]
```

Comment:
because it wouldn't always be that shape. The idea is for the function to either support n_epochs x n_channels x n_times, or just n_channels x n_times. This seemed easier than doing an if/then statement